### PR TITLE
Build all docker images in PR validation

### PR DESCRIPTION
**Pull request by tamirkamara**
# PR for issue #805 

## What is being addressed

All docker images should be built as part of PR validation

## How is this addressed

- Add build steps for all our custom docker images 
- Run tests as part of the docker build for the API. This required moving the python requirements files
- Remove api build steps from the 
- Update python package versions on api/resource-processor to match
- [x] rename the workflow files after this draft is approved
- [ ] update PR checks: add aggregate check and remove Publish Unit Tests Results

### CODE REVIEW

File: .github/workflows/build_and_push_docker_images.yml [hunk 1]
```diff
@@ -17,26 +11,97 @@ concurrency:
...
+        uses: docker/build-push-action@v2
+        with:
+          context: ./resource_processor
+          file: ./resource_processor/vmss_porter/Dockerfile
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+        # TODO: Add testing (#817) before this
+      - name: "Build image: guacamole_server"
+        if: (steps.filter.outputs.api == 'true' || github.event_name == 'workflow_dispatch')
```
Comment (eladiw) [inline .github/workflows/build_and_push_docker_images.yml:70]: fix 'api' typos (all across)

**Review by eladiw — CHANGES_REQUESTED** [review]
typo in the yml file.
looks good other than that

File: .github/workflows/build_and_push_docker_images.yml (hunk 1 above)
Comment (tamirkamara) [reply .github/workflows/build_and_push_docker_images.yml]: done.

**Review by eladiw — APPROVED** [review]
lgtm

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -47,9 +47,9 @@ RUN export PORTER_HOME=/home/vscode/.porter \
 ENV PATH /home/vscode/.porter/:$PATH
 
 # Install requirements
-COPY ["requirements.txt", "requirements-dev.txt", "/tmp/pip-tmp/" ]
```
Comment (TessFerrandez) [inline .devcontainer/Dockerfile:50]: not sure in github how to comment on the move so I'm doing it here... 

I'm a bit confused here... the requirements-dev.txt is supposed to be a more generic python requirements file -- containing things like pre-commit etc. it is not api specific, so it should be in the root... and the root requirements one tied the api and the dev one together (eg. included the api specific and the dev general requirements).  If we have a specific one for the resource processor, the main (root one) should include links to both the dev requirements, the api requirements, and the resource processor requirements - e.g. a one stop shop for all requirements you need for development on your dev machine (installed on docker but also easily installable in a non-docker environment for those who prefer pycharm or other ides).

Also, did this now overwrite the root or the API requirements.txt?

**Review by TessFerrandez — CHANGES_REQUESTED** [review]
Just a question, but want to make sure it gets answered before merging so marking as request changes

File: .github/workflows/build_and_push_docker_images.yml [hunk 3]
```diff
@@ -17,26 +11,97 @@ concurrency:
...
+      - name: "Build image: resource_processor"
+        if: (steps.filter.outputs.resource_processor == 'true' || github.event_name == 'workflow_dispatch')
+        uses: docker/build-push-action@v2
+        with:
+          context: ./resource_processor
+          file: ./resource_processor/vmss_porter/Dockerfile
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+        # TODO: Add testing (#817) before this
```
Comment (TessFerrandez) [inline .github/workflows/build_and_push_docker_images.yml:68]: If we don't have tests here - does this mean that this PR goes untested? Why were they removed?

**Review by TessFerrandez — CHANGES_REQUESTED** [review]

File: .github/workflows/build_and_push_docker_images.yml (hunk 3 above)
Comment (tamirkamara) [reply .github/workflows/build_and_push_docker_images.yml:68]: There're tests but they aren't being triggered as part of the build. Anat is tasked to solve it.

Comment (TessFerrandez) [reply .github/workflows/build_and_push_docker_images.yml:68]: ahh, some other tests, was thinking this was our regular pytests that were removed

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (TessFerrandez) [reply .devcontainer/Dockerfile:50]: per discussion

**Review by TessFerrandez — APPROVED** [review]
LGTM

**Pull request merged**